tests/src/unit_v2_tests.cpp

```diff
@@ -267,4 +272,84 @@ TEST(stronk_units_v2, can_override_units)
+    static_assert(std::same_as<decltype(m * km), meters_squared::value<std::kilo, double>>);
+    static_assert(std::same_as<decltype(m * um), meters_squared::value<std::micro, double>>);
+
+    static_assert(std::same_as<decltype(km * km), meters_squared::value<std::mega, double>>);
+    static_assert(std::same_as<decltype(km * um), meters_squared::value<std::milli, double>>);
+
+    static_assert(std::same_as<decltype(um * um), meters_squared::value<std::pico, double>>);
+
+    static_assert(std::same_as<decltype(m / m), double>);
+    // static_assert(std::same_as<decltype(m / km), double>); // fails compiling on purpose
+    // static_assert(std::same_as<decltype(m / um), double>); // fails compiling on purpose
+
+    auto m_2 = make<meters_squared, double>(1'000.0);
+    auto km_2 = make<std::kilo, meters_squared, double>(1'000.0);
+    auto um_2 = make<std::micro, meters_squared, double>(1'000.0);
+
+    static_assert(std::same_as<decltype(m_2), meters_squared::value<std::ratio<1>, double>>);
+    static_assert(std::same_as<decltype(km_2), meters_squared::value<std::mega, double>>);
+    static_assert(std::same_as<decltype(um_2), meters_squared::value<std::pico, double>>);
+
+    static_assert(std::same_as<decltype(m_2 / m), meters::value<std::ratio<1>, double>>);
+    static_assert(std::same_as<decltype(m_2 / km), meters::value<std::milli, double>>);
+    static_assert(std::same_as<decltype(m_2 / um), meters::value<std::mega, double>>);
+
+    static_assert(std::same_as<decltype(km_2 / m), meters::value<std::mega, double>>);
+    static_assert(std::same_as<decltype(km_2 / km), meters::value<std::kilo, double>>);
+    static_assert(std::same_as<decltype(km_2 / um), meters::value<std::tera, double>>);
+
+    static_assert(std::same_as<decltype(um_2 / m), meters::value<std::pico, double>>);
+    static_assert(std::same_as<decltype(um_2 / km), meters::value<std::femto, double>>);
+    static_assert(std::same_as<decltype(um_2 / um), meters::value<std::micro, double>>);
+
+    // now divide by squared
+    static_assert(std::same_as<decltype(m / m_2), per_meter::value<std::ratio<1>, double>>);
+    static_assert(std::same_as<decltype(km / m_2), per_meter::value<std::kilo, double>>);
+    static_assert(std::same_as<decltype(um / m_2), per_meter::value<std::micro, double>>);
+
+    static_assert(std::same_as<decltype(m / km_2), per_meter::value<std::micro, double>>);
+    static_assert(std::same_as<decltype(km / km_2), per_meter::value<std::milli, double>>);
+    static_assert(std::same_as<decltype(um / km_2), per_meter::value<std::pico, double>>);
+
+    static_assert(std::same_as<decltype(m / um_2), per_meter::value<std::tera, double>>);
+    static_assert(std::same_as<decltype(km / um_2), per_meter::value<std::peta, double>>);
+    static_assert(std::same_as<decltype(um / um_2), per_meter::value<std::mega, double>>);
+}
+
+TEST(stronk_units_v2, scales_are_applied_correctly_when_converted)
```

Comment:
Consider also checking values for some of the multiplications and divisions from the new type test above this one.